src/Enum.php

```diff
@@ -308,7 +310,6 @@ public static function __callStatic($name, $arguments)
      *
      * @return mixed
      * @link http://php.net/manual/en/jsonserializable.jsonserialize.php
-     * @psalm-pure
```

Comment:
Because of this error:

```
ERROR: ImpureMethodCall - src/Enum.php:313:7 - Cannot call an impure constructor from a pure context (see https://psalm.dev/203)
    #[\ReturnTypeWillChange]
```

We should either remove `@psalm-pure` or `#[\ReturnTypeWillChange]`

Since Enum's JSON serialization can be anything, We may keep the second one.